OpenRA.Mods.Common/HitShapes/IHitShape.cs

```diff
@@ -0,0 +1,21 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Graphics;
+
+namespace OpenRA.Mods.Common.HitShapes
+{
+	public interface IHitShape
```

Comment:
Can you please also add implementations for line-vs-shape intersections so that this can play nicely with #9357?